feat(action-release): Add more details for the Sentry Release GitHub action (#12403)
getsentry/sentry-docs: docs/product/releases/setup/release-automation/github-actions/index.mdx

---
title: GitHub Actions
description: "Learn how Sentry and GitHub Actions automate release management, upload source maps and help identify suspect commits."
---

> The [Sentry Release GitHub Action](https://github.com/marketplace/actions/sentry-release) automates Sentry release management in GitHub with just one step. After sending Sentry release information, you’ll be able to associate commits with releases. Additionally, releases are used for applying source maps to view original, un-minified source code.

Reviewer: ```suggestion
The [Sentry Release GitHub Action](https://github.com/marketplace/actions/sentry-release) automates Sentry release management in GitHub with just one step. After sending Sentry release information, you’ll be able to associate commits with releases. Additionally, you can upload source maps to view original, un-minified source code.
```

Reviewer: With debugIds, releases aren't really involved necessarily

For more details about Sentry release management concepts, see the full documentation on [releases](/product/releases/).

## Prerequisites

For the Sentry Release GitHub action to communicate securely with Sentry, you'll need to configure an [Organization Auth Token](https://sentry.io/orgredirect/organizations/:orgslug/settings/auth-tokens/).

> Alternatively, you can also use a [User Auth Token](https://sentry.io/orgredirect/organizations/:orgslug/settings/account/api/auth-tokens/), with the "Project: Read & Write" and "Release: Admin" permissions.

Reviewer: ```suggestion
```
we don't really want to promote these anymore. People should use an org auth token, and just leave it at that

You also need to set your Organization and Project slugs and if you're using a self-hosted Sentry instance, provide the URL used to connect to Sentry via `SENTRY_URL`.

<OrgAuthTokenNote />

```bash
SENTRY_AUTH_TOKEN=___ORG_AUTH_TOKEN___
SENTRY_ORG=___ORG_SLUG___
SENTRY_PROJECT=___PROJECT_SLUG___
# For self-hosted
# SENTRY_URL=https://my-sentry-url
```

> We recommend storing these as [encrypted secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions) on your repository.

Reviewer: these docs from GH, are terrible. Dire need of simplification and a couple simple screenshots.

I'm very tempted that we add an arcade showing someone how to add this secret. This was also very onboxious when setting up CodeCov cc @aj-codecov

Author: Added a screenshot + explanation

  env:
    SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
    SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
    SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
  with:
    environment: production
    version: 'v1.3.4'
```

## Source Maps

If your application is written in JavaScript you probably deploy minified application code. To enable readable
stack traces in your Sentry errors, you need to upload source maps to Sentry.

Provide the path(s) to your source maps via the `sourcemaps` option and set the `inject` flag to allow the GitHub
action to inject information into your source files and source maps to ensure proper un-minification of your stack traces.
Visit [What are Artifact Bundles](/platforms/javascript/sourcemaps/troubleshooting_js/artifact-bundles/) if you want to learn more.

```yml
- uses: actions/checkout@v4
  with:
    fetch-depth: 0

- name: Create Sentry release
  uses: getsentry/action-release@v1
  env:
    SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
    SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
    SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
  with:
    environment: production
    inject: true
    sourcemaps: './dist'
```

<Alert>

>   It's important to deploy the files that have been built on your CI after the Sentry Release GitHub Action

Reviewer: Should we also mention something like, it is also important that this runs after any and all build steps related to bundling, compiling, and/or transpiling?

Author: Yep, good idea.

Author: Updated
